### ISSUE TITLE
Error: Couldn't find UUID for "/media/USER/DRIVE/BACKUP-FOLDER/backintime/PC-NAME/root/1"

### ISSUE THREAD

**Issue by M-W99** [bit-team/backintime#764]
OS: Ubuntu 16.04.2
BackInTime: 1.1.20

When trying to save the config with the "Schedule: When drive get connected (udev)" option selected, it gives me this error:
> Error: Couldn't find UUID for "/media/USER/DRIVE/BACKUP-FOLDER/backintime/PC-NAME/root/1"

Other Schedule options work fine. I'd really like to be able to use this option though so I don't have to leave my drive plugged in all the time.

Thanks!

**Comment by Germar** [comment]
Please run `backintime-qt4 --debug` and change to udev schedule again and post the output from Terminal.

**Comment by M-W99** [comment]
`sudo -i backintime-qt4 --debug`:

> DEBUG: [common/backintime.py:518 arg_parse] Arguments: {'debug': True} | unknownArgs: []
> 
> Back In Time
> Version: 1.1.20
> 
> Back In Time comes with ABSOLUTELY NO WARRANTY.
> This is free software, and you are welcome to redistribute it
> under certain conditions; type `backintime --license' for details.
> 
> DEBUG: [common/backintime.py:585 getConfig] config file: /root/.config/backintime/config
> DEBUG: [common/backintime.py:586 getConfig] profiles: ['1']
> DEBUG: [common/applicationinstance.py:73 GUIApplicationInstance.check] No PID file
> DEBUG: [common/pluginmanager.py:88 PluginManager.load_plugins] Register plugin path /usr/share/backintime/plugins
> DEBUG: [common/pluginmanager.py:104 PluginManager.load_plugins] Add plugin qt4plugin.py
> DEBUG: [common/pluginmanager.py:104 PluginManager.load_plugins] Add plugin userscriptsplugin.py
> DEBUG: [common/pluginmanager.py:104 PluginManager.load_plugins] Add plugin notifyplugin.py
> DEBUG: [common/snapshots.py:953 Snapshots.has_old_snapshots] Found old snapshots: False
> DEBUG: [common/tools.py:594 keyring_supported] No keyring due to import errror.
> DEBUG: [common/applicationinstance.py:73 ApplicationInstance.check] No PID file
> DEBUG: [common/mount.py:50 Mount.__init__] pw-cache is not running
> DEBUG: [common/mount.py:59 Mount.__init__] Call command: /usr/bin/backintime pw-cache start
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/tools.py:594 keyring_supported] No keyring due to import errror.
> DEBUG: [common/tools.py:594 keyring_supported] No keyring due to import errror.
> DEBUG: [common/tools.py:594 keyring_supported] No keyring due to import errror.
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/config.py:375 Config.set_snapshots_path] Check snapshot folder: /media/user/Seagate Backup Plus Drive/BackInTime
> DEBUG: [common/config.py:285 Config.check_config] Check profile Main profile
> DEBUG: [common/tools.py:865 readCrontab] Read 2 lines from users crontab
> DEBUG: [common/config.py:1430 Config.removeOldCrontab] Clearing system Back In Time entries
> DEBUG: [common/config.py:1463 Config.cronLine] Profile: Main profile | Automatic backup: When drive get connected (udev)
> device node not found
> ERROR: [common/config.py:1525 Config.cronLine] Couldn't find UUID for "/media/user/Seagate Backup Plus Drive/BackInTime/backintime/User-HP-Pavilion-dv7/root/1"
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/configfile.py:476 Config.set_current_profile] change current profile: 1
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/applicationinstance.py:88 ApplicationInstance.check] No process with PID 7782
> DEBUG: [common/tools.py:594 keyring_supported] No keyring due to import errror.
> DEBUG: [common/applicationinstance.py:73 ApplicationInstance.check] No PID file
> DEBUG: [common/mount.py:50 Mount.__init__] pw-cache is not running
> DEBUG: [common/mount.py:59 Mount.__init__] Call command: /usr/bin/backintime pw-cache start

**Comment by Germar** [comment]
Please update to current dev version 1.2.0 from ppa:bit-team/testing. This is fixed in there

Duplicate of #607

**Comment by M-W99** [comment]
Ok, it seems to be working but I get this message when plugging in the drive:
> Can't find snapshots folder.
> If it is on a removable drive please plug it.
> Waiting 30 seconds.

I would provide the debug log but I'll have to wait an hour for it to backup again.
Also, are the testing versions safe for actual use? Any risk of backup corruption, etc.?

Thanks!

**Comment by Germar** [comment]
Does your drive mount automatically after plugging in? BiT didn't found the snapshots in the path you configured and will retry for 30sec. Maybe your drive just need a while to come up?

Actually current testing is more stable than current stable. I now have a unittest suite which helped to find and fix couple bugs. I already use testing in production for couple month without problems. But I still like to add some features before releasing current testing.

**Comment by M-W99** [comment]
Yes, it automatically mounts. Perhaps BackInTime detects it before the files are made readable or something? Adding a small delay sounds like it would work, but then again I may be the only one with this issue. It's no big deal really, as the backups seem to work despite the message.

One other issue I've found is that the notifications don't appear to work. I can make a new issue report about this if that would be better, but I'll go ahead and describe it here.

I get the "Can't find snapshots folder..." pop-up when I first connect the drive, but that's it. There's no way to know if the backup has completed or not without having to open BackInTime manually. The backup progress indicator worked on the stable version, but no longer appears to on the testing version. Just a simple "Backup Complete" notification would be perfect!

**Comment by Germar** [comment]
You can add a delay with a `user-callback` script:
```
#!/bin/bash
profile_id="$1"
profile_name="$2"
reason="$3"

case $reason in
    1) #Backup process begins
        sleep 5
        ;;
esac
```

If you run as normal user (not root) there should be a systray icon during taking the snapshot. If not, please create a new bug-report with the output of `backintime --debug backup-job`.

You could add custom notifications with user-callback, too. Take a look at [this example](https://github.com/bit-team/user-callback/blob/master/user-callback.notify)

**Comment by M-W99** [comment]
I'm running as root actually, since it doesn't back up everything in my home folder when run as normal user.

Thanks, I will see what I can do with user-callback.

**Comment by Germar** [comment]
As long as your normal user can read everything inside his own home it should be possible to backup home as user. Only things like /etc need to be backed up with root.

**Comment by M-W99** [comment]
There are actually a few folders such as .gnome and .dbus inside my home folder that I'm not able to read as a normal user, and thus not able to backup when running as a normal user. I realize such folders may not be totally necessary to backup, but I like to have a complete copy.

Is there any particular reason that notifications are disabled for the root version?

**Comment by colinl** [comment]
I believe those folders and contents should be owned by you, I suspect you
may have run something you shouldn't using sudo. I suggest changing the
ownership of them and all contents to yourself. I would not be surprised if
having those owned by root caused strange things to happen.

Colin

**Comment by Germar** [comment]
I totally agree with Colin! You can check which folders are not owned by you with
```
sudo find ~/ ! -user $(whoami) -ls
```
and fix the ownership with
```
sudo find ~/ ! -user $(whoami) -exec chown $(whoami) {} +
```

**Comment by M-W99** [comment]
Ok, thank you both. I have changed the ownership of all the files owned by root in my home folder.

I'm currently using my Pictures folder to test with, and after switching to the normal user version the tray icon appears when running a manual backup, but I'm not seeing any notification at all when the udev backup runs.
Again, the backup completes successfully because I can open BackInTime and see it in the list with the time stamp as the time when I plugged the drive in, but I get no notification at all.
